packages/netlify-cms-core/src/reducers/editorialWorkflow.ts

```diff
@@ -121,13 +140,48 @@ const unpublishedEntries = (state = Map(), action: EditorialWorkflowAction) => {
       );
 
     case UNPUBLISHED_ENTRY_PUBLISH_SUCCESS:
+      return state.withMutations(map => {
+        map.deleteIn(['entities', `${action.payload!.collection}.${action.payload!.slug}`]);
+        map.deleteIn([
+          'entities',
+          `${action.payload!.collection}.${action.payload!.slug}`,
+          'isPublishing',
+        ]);
+      });
+
     case UNPUBLISHED_ENTRY_PUBLISH_FAILURE:
       return state.withMutations(map => {
         map.deleteIn(['entities', `${action.payload!.collection}.${action.payload!.slug}`]);
+        map.setIn(
+          ['entities', `${action.payload!.collection}.${action.payload!.slug}`, 'isPublishing'],
+          false,
+        );
       });
 
+    case UNPUBLISHED_ENTRY_DELETE_REQUEST:
```

Comment:
Where is the `isDeleting` state used?

Reply:
In UNPUBLISHED_ENTRY_DELETE_REQUEST and UNPUBLISHED_ENTRY_DELETE_SUCCESS cases we set `isDeleting`. So to be consistent I thought I should set `isDeleting` as false.

Reply:
I think `UNPUBLISHED_ENTRY_DELETE_REQUEST` didn't modify the state, so I don't see a reason to add it at the moment.
We can keep only the `UNPUBLISHED_ENTRY_DELETE_SUCCESS` state change I think.